### PR TITLE
Maps api

### PR DESCRIPTION
**Pull request by chriswhong**
@nerik can you take a look at this new module for using named maps?

### CODE REVIEW

File: examples/maps-named/named.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+var CartoDB = require('../..');
```
Comment (nerik) [inline examples/maps-named/named.js]: you can remove this file I think?

File: lib/maps.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+util.inherits(Named, Stream);
+
+
+function Named(config) {
+
+  for(var i in config){
+    this[i] = config[i];
+  }
+
+  this.maps_api_url = this.maps_api_url || Mustache.render(resources.maps_api_url, config);
```
Comment (nerik) [inline lib/maps.js]: This code is the same as in SQL and Import constructors. Would be nice to inherit from some base class.

File: examples/maps-named/named.js (hunk 1 above)
Comment (chriswhong) [inline examples/maps-named/named.js]: This is the working example... shouldn't we keep it in?

File: lib/maps.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  return this;
+}
+
+Named.prototype.create = function(options) {
+  debug('POSTing your template to %s', this.maps_api_url);
+
+  var self = this;
+  var promise = new _Promise();
+
+  request.post({
```
Comment (nerik) [inline lib/maps.js]: How about factoring all the request calls in some utility method?
```
Named.prototype._getRequestPromise = function(method, body, url, callback) {
  return request[method]({
    url: url,
    body: body,
    qs: ...
  }, callback
}
```
could even be on a separate utils module.

File: examples/maps-named/named.js (hunk 1 above)
Comment (nerik) [inline examples/maps-named/named.js]: Sry, thought it was all commented out.
You can safely remove comments. There's always a way to retrieve history with git.

**Pull request merged**